kronos/src/androidMain/kotlin/com/softartdev/kronos/AndroidNetworkClock.kt

```
@@ -1,26 +1,25 @@
 package com.softartdev.kronos
 
+import com.instacart.truetime.time.TrueTime
 import com.instacart.truetime.time.TrueTimeImpl
 import com.instacart.truetime.time.TrueTimeParameters
 
-private const val SYNC_INTERNAL_IN_MILLIS = 60_000L
-private const val NTP_HOST_CLOUD_FLARE = "time.cloudflare.com"
-private const val NTP_HOST_GOOGLE = "time.google.com"
-
 object AndroidNetworkClock : NetworkClock {
-    private val trueTime = TrueTimeImpl(
-        params = TrueTimeParameters.Builder()
-            .ntpHostPool(arrayListOf(
-               NTP_HOST_CLOUD_FLARE,
-                NTP_HOST_GOOGLE
-            ))
-            .syncIntervalInMillis(SYNC_INTERNAL_IN_MILLIS)
-            .buildParams()
-    )
+    lateinit var trueTime: TrueTime
 
-    fun sync() {
+    override fun sync(pool: String, syncIntervalInMilliSeconds: Long) {
+        trueTime = TrueTimeImpl(
+            params = TrueTimeParameters.Builder()
+                .ntpHostPool(arrayListOf(pool))
+                .syncIntervalInMillis(syncIntervalInMilliSeconds)
+                .buildParams()
+        )
         trueTime.sync()
     }
 
-    override fun getCurrentNtpTimeMs(): Long = trueTime.now().time
+    override fun getCurrentNtpTimeMs(): Long? = try {
+        trueTime.nowTrueOnly().time
+    } catch (_: Exception) {
+        null
```

why are we silently ignoring an exception now?